doc/design/embargo-redesign.md

```diff
@@ -116,3 +116,49 @@ sequenceDiagram
+
+   On the other hand, perhaps scientists would have a reason to want to include the same data in both an embargoed Dandiset and an open one. For instance, perhaps embargoed experiments are using some base but non-secret data within an embargoed Dandiset, and that same data needs to be part of an open Dandiset.
+
+2. **Allow this type of sharing but force the user to provide an explicit flag to do so.** In this remediation, the user would receive a warning that they are trying to include embargoed data in an open Dandiset. The API / client would return an error with a prompt to retry with a `--force` type flag, which would then allow the upload to occur. The embargoed assets would remain hidden to the public in the S3 bucket, and data access to the open Dandiset’s asset holding would occur via presigned URLs (the same way as for authorized users to embargoed Dandiset assets).
+
+The major questions here are:
+
+- **Are the semantics valid of placing an embargoed asset in an open Dandiset?**
+- **Would a scientist ever want to push data to an embargoed Dandiset, and also to an open Dandiset?**
+
+### Two embargoed Dandisets include a shared embargoed asset
+
+**Scenario.** Someone creates embargoed Dandisets 5 and 6, uploading asset C to each. Then, they unembargo Dandiset 6, leading to a scenario similar to the previous one.
+
+**Quirk.** This is the same state as the previous scenario, but the same remediations do not seem to apply. Disallowing the unembargo operation is a nonstarter because that leads to a sort of deadlock: there would be no way to unembargo Dandisets 5 and 6. And forcing the user to explicitly include a flag does not help since we still end up in that deadlock situation. Furthermore, the questions about intent do not come up here: it seems reasonable that a scientist would indeed prepare two Dandisets with a shared, embargoed asset.
+
+**Remediation.** In this case, it seems that we need to allow for unembargo and fall back to the situation described above, where access to the asset is granted through presigned URLs, without otherwise disclosing the fact that the asset is still present in another embargoed Dandiset. A warning to the user on their attempt to unembargo may be warranted so they can decide if they want to expose that asset while it is still under embargo in the other Dandiset.
+
+Finally, this brings up a question about when to remove the `embargoed` tag from the object in the S3 bucket: we would insert logic to detect whether the embargoed asset is included in any other embargoed Dandisets, and only trigger the tag removal if there are none. This brings up one major question:
+
+- **Do these situations nullify the effect/need to keep data hidden in the bucket at all?**
+The answer to this is probably “no”, but some discussion may help to illuminate answers to the other questions and quirks above.
```

Comment:
those are great user stories!  Thank you @waxlamp .  I think we should just generalize them into some concise "design decision" which would inform on what is to happen in any of them as a result/remediation.  Echoing the first one I would say it could be

- Anyone with access to EMBARGOED content can make it OPEN
  - I think this would be the crucial statement which would offload responsibility to the owner (or someone with access) to embargoed data.  May be a suboptimal example, but it is similar to "leaking the data" -- as soon as someone leaks private data, it becomes known publicly.  The same here -- if someone with access to EMBARGOED data decides to upload it publicly, they can.
  - We indeed would want interface(s) to make such transitions from EMBARGOED to OPEN to be explicit. I see two "entry points"
     - Explicit "make content of a dandiset (or prefix within it? e.g. a participant)" OPEN -- that goes to "Object Tag Removal Workflow".
     -  Make content OPEN via upload.
  - If we do not assume that checksums are private data -- we would need to verify that the person indeed has access to EMBARGOED content to make it OPEN upon upload. We would need a (missing ATM) workflow here which instead of minting a new blob would just make EMBARGOED to become OPEN upon verification of upload digest.

- EMBARGOED dandiset can point to OPEN content
- EMBARGOED content can be made OPEN by any authorized user
- Only a dandiset without assets pointing to EMBARGOED content can be announced to be OPEN
   - so, explicit action: UNEMBARGOING of dandiset consists of unembargoing any still embargoed content first
   - UNEMBARGOING some content might make one or more dandisets available to transition to OPEN without unembargoing more conent
- Upload to EMBARGOED dandiset makes new to the archive content EMBARGOED
- Upload to OPEN dandiset makes content (new or already known to archive) OPEN
  - that is where we might want to indeed add some explicit option to allow for unembargoing of embargoed content, since otherwise people might get a little pissed after someone from their group would "leak" the embargoed data.

But in general it boils down to be able to inform user on the effects of potential unembargo -- would we be able to say how many dandisets would be affected? how many would become OPEN (as in content)?

Reply:
> We indeed would want interface(s) to make such transitions from EMBARGOED to OPEN to be explicit. I see two "entry points"
> - Explicit "make content of a dandiset (or prefix within it? e.g. a participant)" OPEN -- that goes to "Object Tag Removal Workflow".
> - Make content OPEN via upload.

I think the second option is the right one. It is simpler, and it leaves the idea of "embargoed Dandiset" whole (indeed, the other way to make assets in an embargoed Dandiset OPEN is, quoting from your response, *so, explicit action: UNEMBARGOING of dandiset consists of unembargoing any still embargoed content first*, which also considers "embargo" to be a state on a whole Dandiset).

Everything else you wrote, I believe I understand and agree with. I will rework the PR text to reflect what you've said here and we can do a review round to make sure we agree on how it all should work.

Reply:
I might be misunderstanding you saying "the right one" here. I do not think of these two options as mutually exclusive alternatives - rather they seems to be the two ways we should allow for making content OPEN.  Are you suggesting it ('make content OPEN via upload') to be the only way to make content OPEN?

Reply:
> Are you suggesting it ('make content OPEN via upload') to be the only way to make content OPEN?

This is indeed what I was suggesting. My reasons for this:
- simplifies the mental model for the user--in particular, the user has less to worry about in terms of managing what is open and what is hidden
- keeps the concept of embargo simpler: a Dandiset is either embargoed or not, and upon performing the "unembargo" action, all contained assets become open

Much of this comes from trying to control complexity, even if in the future we are going to move to that hairier model where any kind of assets can be present in any Dandiset (and the simplicity of a binary embargo state becomes fuzzier). So I may need to compromise here, but the cost is much greater complexity in what we're attempting to build (I just need to be open about this tradeoff).

Reply:
I don't think that would work since 
- it would pretty much bring back the original motivation for redesign - problem of Inefficiency , with all issues just offloaded to the uploader instead of our automations which now need to move data between buckets
- as a result, would make "UNEMBARGO"ing a very time consuming and unpleasant experience thus demotivating users from making their data OPEN.

I think I could argue also that the first reason you stated would not really be fulfilled, and the 2nd one remains pertinent to the proposed first "entry point" to unembaroing content - done directly on the archive:

> * Explicit "make content of a dandiset (or prefix within it? e.g. a participant)" OPEN -- that goes to "Object Tag Removal Workflow".

So may be we are not fully understanding each other here.

Reply:
Yes I think we must be miscommunicating a bit. Among other things, I'm not sure how what I proposed causes any performance problems. What I'm suggesting is something like this:
- uploading an asset to an open dandiset would cause it to become unhidden in the bucket
- clicking an "unembargo" button would cause all of an embargoed dandiset's (hidden) assets to become unhidden (and the status of the dandiset becomes `open`)

Both of these "entry points" would work with the redesigned embargo concept (i.e., no performance issues). Allowing users to unhide assets one by one would also be compatible with this way of doing things, now that I think about it (it is equivalent to uploading that asset somewhere else), but I guess I don't see why any user would do so--but perhaps there is a reason (maybe some of the example use cases Satra posted in a different thread).

In any case, maybe we can discuss this face to face.